testing/run

```diff
@@ -144,15 +153,12 @@ fi
 
```

Comment:
@sagidayan, could you try to find a way to avoid this defaulting?

you have access to the OSD cluster at this stage, so default to `osde2e/gpu-addon` only if ...
1.  `$(hostname) == `name of the test pod` 
2. or, the cluster is an OSD
3. or, the script is running from the CI image (`$INSIDE_CI_IMAGE == "y"`)
3. or, create a dedicated image for OSD testing and add a specific flag (`INSIDE_OSD_IMAGE="y`")

Reply:
Added a HOSTNAME check. Sets all the osde2e defaults